src/Resources/js/modules/participation-update-handler.ts

```diff
@@ -19,192 +22,418 @@ export interface AcceptOfferData {
+        this.updateCheckboxWrapper($checkbox);
+
+        ParticipationUpdateHandler.setCount($checkbox, state, status.count);
     }
 
-    public static changeToOfferIsAvailable($checkbox: JQuery, url: string) {
-        // change
-        this.changeCheckboxAttributes($checkbox, ParticipationAction.ACCEPT_OFFER, url);
-        let participantCounter = $checkbox.data(ParticipantCounter.NAME);
-        if (ParticipationState.OFFER_AVAILABLE !== participantCounter.getParticipationState()) {
-            this.changeParticipationCounter($checkbox, ParticipationState.OFFER_AVAILABLE);
+    private static setCount($checkbox: JQuery, state: State, count?: number): void {
+        let $countContainer = $checkbox.closest('.wrapper-meal-actions').find('.participants-count');
+        if (1 > $countContainer.length) {
+            return;
         }
 
-        // update
-        this.updateCheckboxEnabled($checkbox);
-        this.updateCheckBoxWrapper($checkbox);
-        this.toggleTooltip($checkbox, TooltipLabel.AVAILABLE_MEAL);
+        $countContainer.removeClass().addClass('participants-count');
+
+        switch (state) {
+            case State.OPEN:
+            case State.BOOKED:
+                $countContainer.addClass('participation-allowed');
+                break;
+            case State.OFFERED:
+                $countContainer.addClass('participation-pending');
+                break;
+        }
+
+        if (undefined === count) {
+            return;
+        }
+
+        let partCount = $countContainer.find('.count');
+        if (0 < partCount.length) {
+            partCount.text(count);
+        }
     }
 
-    public static changeToSwapState($checkbox: JQuery, url: string, participantsCount?: number) {
-        // change
-        $checkbox.prop('checked', true);
-        this.changeCheckboxAttributes($checkbox, ParticipationAction.SWAP, url);
-        this.changeParticipationCounter($checkbox, ParticipationState.DEFAULT, participantsCount);
+    private static toggleTooltip($checkbox: JQuery, label?: TooltipLabel) {
+        let $tooltip = $checkbox.closest('.wrapper-meal-actions').find('.tooltiptext');
+        if (undefined !== label) {
+            $.getJSON('/labels.json')
+                .done(function (labels: Labels) {
+                    if ('de' === $('.language-switch').find('span').text()) {
+                        $tooltip.text(labels.de.tooltip[label]);
+                    } else {
+                        $tooltip.text(labels.en.tooltip[label]);
+                    }
+                    $tooltip.toggleClass('active');
+                });
+        } else {
+            $tooltip.toggleClass('active');
+        }
+    }
 
-        // update
-        this.updateCheckboxEnabled($checkbox);
-        this.updateCheckBoxWrapper($checkbox);
-        this.toggleTooltip($checkbox);
+    private static updateCheckboxWrapper($checkbox: JQuery) {
+        let $checkboxWrapper = $checkbox.closest('.checkbox-wrapper');
```

Comment:
why not write like this?

$checkbox.closest('.checkbox-wrapper')
       .toggleClass('checked', $checkbox.is(':checked'))
       .toggleClass('disabled', $checkbox.is(':disabled'));

Reply:
Updated